runtime/gc_vlhgc/ConfigurationIncrementalGenerational.cpp

```diff
@@ -203,9 +202,9 @@ MM_ConfigurationIncrementalGenerational::createDefaultMemorySpace(MM_Environment
 	MM_HeapRegionManager *regionManager = extensions->heapRegionManager;
 	Assert_MM_true(NULL != regionManager);
 
-	/* Create Sweep Pool Manager for MemoryPoolBumpPointer */
-	extensions->sweepPoolManagerBumpPointer = MM_SweepPoolManagerVLHGC::newInstance(MM_EnvironmentVLHGC::getEnvironment(env));
-	if (NULL == extensions->sweepPoolManagerBumpPointer) {
+	/* Create Sweep Pool Manager for MPAOL */
+	extensions->sweepPoolManagerAddressOrderedList = MM_SweepPoolManagerAddressOrderedList::newInstance(env);
```

Comment:
I don't yet understand what are specific changes are required for SweepPoolManager, but I think we should continue to use MM_SweepPoolManagerVLHGC which would be a subclass of SweepPoolManagerAddressOrderedList.
The instance would be called sweepPoolManagerVLHGC.

We would introduce some virtual methods that would be different for MM_SweepPoolManagerVLHGC  vs SweepPoolManagerAddressOrderedList.

I'll try to be more specific, once I understand the changes better.